cli/ssh.go

```diff
@@ -74,24 +72,23 @@ func (r *RootCmd) ssh() *clibase.Cmd {
 					logger.Error(ctx, "command exit", slog.Error(retErr))
 				}
 			}()
-			if logToFile {
-				// we need a way to ensure different ssh invocations don't clobber
-				// each other's logs. Date-time strings will likely have collisions
-				// in unit tests and/or scripts unless we extend precision out to
-				// sub-millisecond, which seems unwieldy.  A simple 5-character random
-				// string will do it, since the operating system already tracks
-				// dates and times for file IO.
-				qual, err := cryptorand.String(5)
+			if logFilePath != "" {
+				logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0o600)
 				if err != nil {
-					return xerrors.Errorf("generate random qualifier: %w", err)
+					return xerrors.Errorf("error opening %s for logging: %w", logFilePath, err)
 				}
-				logPth := path.Join(logDir, fmt.Sprintf("coder-ssh-%s.log", qual))
-				logFile, err := os.Create(logPth)
-				if err != nil {
-					return xerrors.Errorf("error opening %s for logging: %w", logPth, err)
-				}
-				logger = slog.Make(sloghuman.Sink(logFile))
-				defer logFile.Close()
+				// HACK: Something was keeping a reference to this file
+				// after the goroutine ends, leading to the race observed
+				// here: https://github.com/coder/coder/actions/runs/5178818818/jobs/9331016395.
```

Comment:
the race is `watchAndClose()` is run on a separate goroutine, and it attempts to log while we are closing the log file.  A non-hack solution is to wait for `watchAndClose()` to finish (e.g. a wait group) before closing the log file.

Reply:
Ah ok, not sure why I didn't catch those 🤦🏽‍♂️. Fixed in ad2c946290107faf01c15d1aa133d0cca903b194.